### PR TITLE
SAAS-11985: fix getImportantValues

### PR DESCRIPTION
**Pull request by edenhassid**
fix getImportantValues

---

_Additional context for reviewer_

---
_Release Notes_: 
_Replace me with a short sentence that describes the effect of this change on Salto users_

---
_User Notifications_: 
_Replace me with a short sentence that describes changes that will appear in NaCls and are not caused by user actions (e.g. a new annotation, field values that are converted to references, etc). Hidden changes should not be listed._

### CODE REVIEW

File: packages/adapter-utils/src/important_values.ts [hunk 1]
```diff
@@ -127,9 +128,19 @@ export const getImportantValues = async ({
     return extractImportantValuesFromElement({ importantValues, element, indexedOnly, highlightedOnly })
   }
   if (isField(element) || isInstanceElement(element)) {
-    try {
+    const getTypeObj = async (): Promise<ObjectType> => {
       const typeObj = await element.getType(elementSource)
-      const importantValues = typeObj?.annotations[CORE_ANNOTATIONS.IMPORTANT_VALUES]
+      return isPlaceholderObjectType(typeObj) ? elementSource?.get(typeObj.elemID) : typeObj
```
Comment (netama) [inline packages/adapter-utils/src/important_values.ts:133]: nit: maybe add a comment explaining that we reach this flow for types that are hidden?

File: packages/adapter-utils/src/important_values.ts [hunk 2]
```diff
@@ -127,9 +128,19 @@ export const getImportantValues = async ({
...
+    }
+    try {
+      const typeObj = await getTypeObj()
+      if (typeObj === undefined) {
+        log.trace(
+          `could not get important values as type is undefined returning [] for element ${element.elemID.getFullName()}`,
```
Comment (netama) [inline packages/adapter-utils/src/important_values.ts:139]: tiny nit:
```suggestion
          `could not get important values as type is undefined, returning [] for element ${element.elemID.getFullName()}`,
```

File: packages/adapter-utils/test/important_values.test.ts [hunk 3]
```diff
@@ -160,6 +161,30 @@ describe('getImportantValues', () => {
       { key: 'doesNotExist', value: undefined },
     ])
   })
+  it('should get the right important values for an instance when obj is', async () => {
```
Comment (netama) [inline packages/adapter-utils/test/important_values.test.ts:164]: nit: unfinished sentence?

**Review by netama — APPROVED** [review]

File: packages/adapter-utils/test/important_values.test.ts (hunk 3 above)
Comment (edenhassid) [reply packages/adapter-utils/test/important_values.test.ts]: missed it 😅

**Pull request merged**